### PR TITLE
AP-922: Invalid but focused input fields, (Multi)Selectors, Comboboxes etc. should have a box shadow to indicate focus

### PR DESCRIPTION
**Pull request by misicnenad**
## Explanation of the solution

Added some missing input styles for different states (see `ImgEditor`, `RichTextEditor`, `CheckField`).

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp

## UI changes for review

Regular field:
- error:
image
- error + focus:
image

TokenField:
- error:
image
- error + focus:
image

MultiSelector:
- error:
image
- before (focused):
image
- new error + focus:
image

ImgEditor:
- before:
image
- new error:
image
- error + focus:
image

RichTextEditor:
- before:
image
- new error:
image
- error + focus:
image

CheckField (note: this is not this field's actual behavior, updated the schema just to force this field error to appear)
- before:
image
- new error:
image
- before adding focus shadow:
image
- new error + focus:
image

EDIT: removed `CountrySelector` changes

### CODE REVIEW

File: src/components/Selector/FocusableInput.tsx [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
-      aria-hidden
-      className="peer h-0 w-0 focus:outline-none absolute"
-      tabIndex={-1}
-    />
-  );
-});
+const FocusableInput = forwardRef<HTMLInputElement>(
+  function Input(_props, ref) {
+    return (
+      <input ref={ref} aria-hidden className="h-0 w-0 absolute" tabIndex={-1} />
```
Comment (misicnenad) [inline src/components/Selector/FocusableInput.tsx:6]: no need for `peer` and `focus:` styling as the field is not focusable

File: src/index.css [hunk 2]
```diff
@@ -299,12 +302,6 @@ html:is(.dark) [data-amplify-authenticator] {
     @apply field-input;
   }
 
-  /* This css is required to handle createElement error UI  */
-  .field > textarea[aria-invalid="true"],
-  .field > input[aria-invalid="true"] {
-    @apply border-red;
```
Comment (misicnenad) [inline src/index.css:305]: `field`'s border should not change based on child input's state -> only the child input's border should change, which is handled with `field-input` styling (which is by extension applied automatically to all `.field > textarea, .field > input` components, see above)

File: src/components/Selector/MultiSelector.tsx [hunk 3]
```diff
@@ -66,7 +66,7 @@ export function MultiSelector<
           aria-invalid={!!get(errors, name)?.message}
           aria-disabled={isDisabled}
           as="div"
-          className={`${button} ${styles.selectorButton} p-1 focus-within:border-gray-d1 focus-within:dark:border-blue-l2`}
```
Comment (ap-justin) [inline src/components/Selector/MultiSelector.tsx:69]: bug: tabbing into `<Multiselector/>` doesn't focus
repro: 
go to registration contact details and tab from website to sdg
image

**Review by ap-justin — CHANGES_REQUESTED** [review]

File: src/components/Selector/MultiSelector.tsx [hunk 4]
```diff
@@ -66,7 +66,7 @@ export function MultiSelector<
...
+          className={`${button} ${styles.selectorButton} p-1`}
```
Comment (ap-justin) [inline src/components/Selector/MultiSelector.tsx:69]: bug: when selecting countries , focus border now is not visible

(master: focused when selecting countries)
image

**Review by ap-justin — CHANGES_REQUESTED** [review]

**Review by ap-justin — CHANGES_REQUESTED** [review]

File: src/components/CountrySelector/CountrySelector.tsx [hunk 5]
```diff
@@ -52,7 +52,7 @@ export default function CountrySelector<
       value={country}
       onChange={onCountryChange}
       as="div"
-      className={`relative items-center grid grid-cols-[auto_auto_1fr] w-full field-container ${
+      className={`relative items-center flex w-full field-container ${
```
[the remark this answers is not in the document]
Comment (misicnenad) [reply src/components/CountrySelector/CountrySelector.tsx:55]: Reverted, created ticket for this 
https://linear.app/angel-protocol/issue/AP-962/user-should-be-able-to-click-anywhere-within-countryselector-to-focus

File: src/components/Selector/MultiSelector.tsx (hunk 4 above)
Comment (misicnenad) [reply src/components/Selector/MultiSelector.tsx]: Fixed https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2568/commits/72381756f65dbb2cff34f58cd45960eb88baad71

File: src/components/Selector/MultiSelector.tsx (hunk 3 above)
Comment (misicnenad) [reply src/components/Selector/MultiSelector.tsx:69]: Fixed https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2568/commits/72381756f65dbb2cff34f58cd45960eb88baad71

**Review by SovereignAndrey — APPROVED** [review]

**Pull request merged**